### PR TITLE
Add --devices flag to bud and from

### PR DESCRIPTION
**Pull request by rhatdan**
Some Dockerfiles (fuse-overlay) require additional devices to be in the
build environment.

This patch allows the user to specify additional devices.

Also I noticed that CapAdd and CapDrop was not working in buildah bud situations,
so this patch also fixes this.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: pkg/parse/parse.go [hunk 1]
```diff
@@ -886,3 +888,71 @@ func RegistryFromFullName(input string) string {
 	}
 	return split[0]
 }
+
+// ParseDevice parses device mapping string to a src, dest & permissions string
+func ParseDevice(device string) (string, string, string, error) { //nolint
```
Comment (vrothberg) [inline pkg/parse/parse.go]: What was the linter complaining about?

Comment (vrothberg) [reply pkg/parse/parse.go]: I would love to have some unit tests.

File: pkg/parse/parse.go [hunk 2]
```diff
@@ -886,3 +888,71 @@ func RegistryFromFullName(input string) string {
...
+
+	if dst == "" {
+		dst = src
+	}
+	return src, dst, permissions, nil
+}
+
+// IsValidDeviceMode checks if the mode for device is valid or not.
+// IsValid mode is a composition of r (read), w (write), and m (mknod).
+func IsValidDeviceMode(mode string) bool {
```
Comment (vrothberg) [inline pkg/parse/parse.go]: I would love to have some unit tests.

Comment (vrothberg) [reply pkg/parse/parse.go]: Do we need libpod to call this func? We could make it private .

File: tests/bud.bats [hunk 3]
```diff
@@ -1641,3 +1641,12 @@ load helpers
   touch "$DIR"/Dockerfile
   run_buildah 0 bud "$DIR"/Dockerfile
 }
+
+@test "bud with additional device" {
+  target=alpinepriv
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --device /dev/fuse -t ${target} -f ${TESTSDIR}/bud/device/Dockerfile ${TESTSDIR}/bud/device
```
Comment (vrothberg) [inline tests/bud.bats]: To avoid potential regressions, can we also try that with different modes (rwm)?

**Comment by rhatdan** [comment]
@rh-atomic-bot r=vrothberg

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@jlebon Is Homu hung?

**Comment by jlebon** [comment]
Yeah, you can give it a jog with:
@rh-atomic-bot retry

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot r=vrothberg

**Comment by jlebon** [comment]
It's failing because of https://github.com/containers/buildah/pull/1840#issuecomment-528980890. Should be able to just bump it back down to `29` for now.

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot r=@TomSweeneyRedHat

**Pull request closed without merging**